### PR TITLE
Record: Allow to configure Perf Events based on Model

### PR DESCRIPTION
**Pull request by janaknat**
Icelake and Sapphire Rapids have different configurations for the PMU events. Change them based on the Model during run-time.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/data/intel_perf_events.rs [hunk 1]
```diff
@@ -39,7 +39,7 @@ static L1_INSTRUCTIONS: NamedTypeCtr = NamedTypeCtr {
 static BACKEND_STALLS: NamedTypeCtr = NamedTypeCtr {
     perf_type: PerfType::RAW,
     name: "Backend-Stalls",
-    config: 0x10a2,
```
Comment (lrbison) [inline src/data/intel_perf_events.rs:42]: correcting a bug too?  I didn't see it mentioned in commit message.

File: src/data.rs [hunk 2]
```diff
@@ -4,8 +4,10 @@ pub mod diskstats;
 pub mod flamegraphs;
 #[cfg(target_arch = "aarch64")]
 pub mod grv_perf_events;
+pub mod intel_icelake_perf_events;
```
Comment (RamaMalladiAWS) [inline src/data.rs:7]: this should go after line 8 below.

File: src/data.rs [hunk 3]
```diff
@@ -4,8 +4,10 @@ pub mod diskstats;
...
 #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
 pub mod intel_perf_events;
```
Comment (RamaMalladiAWS) [inline src/data.rs:9]: can you add a comment as to which Intel architecture does this file apply to?

File: src/data/intel_icelake_perf_events.rs [hunk 4]
```diff
@@ -0,0 +1,40 @@
+use crate::data::perf_stat::{NamedCtr, NamedTypeCtr, PerfType};
+
+static CYCLES: NamedTypeCtr = NamedTypeCtr {
```
Comment (RamaMalladiAWS) [inline src/data/intel_icelake_perf_events.rs:3]: These are same as the default in intel_perf_events.rs. So, does this architecture specific config file provide delta counters or all?

**Review by RamaMalladiAWS — COMMENTED** [review]
General comments, 1 change request.

File: src/data/intel_icelake_perf_events.rs (hunk 4 above)
Comment (janaknat) [reply src/data/intel_icelake_perf_events.rs:3]: We make sure the nr and dr are in the same file. We don't want to refer to other files for a PMU event.

File: src/data.rs (hunk 2 above)
Comment (janaknat) [reply src/data.rs]: Yeah. Changing this as it should only be compiled on x86. rustfmt moved these :( .

**Pull request merged**